okta/services/governance/resource_request_condition.go

```diff
@@ -0,0 +1,571 @@
+		accessDurationSettings.Type = types.StringValue(settings.AccessDurationSettingsRequesterSpecifiedDuration.Type)
+		accessDurationSettings.Duration = types.StringValue(settings.AccessDurationSettingsRequesterSpecifiedDuration.MaximumDuration)
+		return &accessDurationSettings
+	}
+	return nil
+}
+
+func setAccessScopeSettings(settings governance.AccessScopeSettingsFullAccessScopeSettings) (*Settings, diag.Diagnostics) {
+	var setting Settings
+	if settings.GroupAccessScopeSettings != nil {
+		setting.Type = types.StringValue(settings.GroupAccessScopeSettings.GetType())
+		groupIds := settings.GroupAccessScopeSettings.GetGroups()
+		var ids []IdModel
+		for _, groupId := range groupIds {
+			idModel := IdModel{
+				Id: types.StringValue(groupId.GetId()),
+			}
+			ids = append(ids, idModel)
+		}
+		setting.Ids = ids
+	} else if settings.EntitlementBundleAccessScopeSettings != nil {
+		setting.Type = types.StringValue(settings.EntitlementBundleAccessScopeSettings.GetType())
+		entitlementBundleIds := settings.EntitlementBundleAccessScopeSettings.GetEntitlementBundles()
+		var ids []IdModel
+		for _, entitlementBundleId := range entitlementBundleIds {
+			idModel := IdModel{
+				Id: types.StringValue(entitlementBundleId.GetId()),
+			}
+			ids = append(ids, idModel)
+		}
+		setting.Ids = ids
+	} else if settings.ResourceDefaultAccessScopeSettings != nil {
+		setting.Type = types.StringValue(settings.ResourceDefaultAccessScopeSettings.GetType())
+	}
+	return &setting, nil
+}
+
+func setRequesterSettings(settings governance.RequesterSettingsFullRequesterSettings) (*Settings, diag.Diagnostics) {
+	var setting Settings
+	if settings.GroupsRequesterSettings != nil {
+		setting.Type = types.StringValue(settings.GroupsRequesterSettings.GetType())
+		groupIds := settings.GroupsRequesterSettings.GetGroups()
+		var ids []IdModel
+		for _, groupId := range groupIds {
+			idModel := IdModel{
+				Id: types.StringValue(groupId.GetId()),
+			}
+			ids = append(ids, idModel)
+		}
+		setting.Ids = ids
+	} else if settings.EveryoneRequesterSettings != nil {
+		setting.Type = types.StringValue(settings.EveryoneRequesterSettings.GetType())
+	}
+	return &setting, nil
+}
+
+func createRequestCondition(data requestConditionResourceModel) (*governance.RequestConditionCreatable, diag.Diagnostics) {
+	req := governance.RequestConditionCreatable{}
+	req.Name = data.Name.ValueString()
+	req.ApprovalSequenceId = data.ApprovalSequenceId.ValueString()
+	if !data.Description.IsNull() && data.Description.ValueString() != "" {
+		req.Description = data.Description.ValueStringPointer()
+	}
+	if !data.Priority.IsNull() {
+		req.Priority = data.Priority.ValueInt32Pointer()
+	}
+
+	accessScopeSettings := governance.AccessScopeSettingsCreatableAccessScopeSettings{}
+	if data.AccessScopeSettings.Type.ValueString() == "GROUPS" {
+		accessScopeSettings.AccessScopeSettingsCreatableGroupAccessScopeSettings.Type = "GROUPS"
```

Comment:
@aditya-okta missing the same check you have below.

```suggestion
if accessScopeSettings.AccessScopeSettingsCreatableGroupAccessScopeSettings == nil {
	accessScopeSettings.AccessScopeSettingsCreatableGroupAccessScopeSettings = &governance.AccessScopeSettingsCreatableGroupAccessScopeSettings{}
}
accessScopeSettings.AccessScopeSettingsCreatableGroupAccessScopeSettings.Type = "GROUPS"
```